### ISSUE TITLE
Hibernate 6 upgrade: native query registration

### ISSUE THREAD

**Issue by sguilhen** [keycloak/keycloak#16330]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

storage

### Describe the bug

After upgrading to hibernate 6, the query registration code in both legacy and map stores needs to be revisited. In particular, `JpaMapUtils` and `JpaUtils` have the following piece of code in `configureNamedQuery`:

```
        if (isNative) {
            // deprecation notes mention usage of a new QueryEngine in hibernate 6
//            NativeSQLQuerySpecification spec = new NativeSQLQuerySpecification(querySql, new NativeSQLQueryReturn[0], Collections.emptySet());
//            sessionFactory.getQueryPlanCache().getNativeSQLQueryPlan(spec);
            sessionFactory.addNamedQuery(queryName, entityManager.createNativeQuery(querySql));
        } else {
//            sessionFactory.getQueryPlanCache().getHQLQueryPlan(querySql, false, Collections.emptyMap());
            sessionFactory.addNamedQuery(queryName, entityManager.createQuery(querySql));
        }
```

`NativeSQLQuerySpecification`  and `NativeSQLQueryReturn` types are no longer available. Migration notes in the Hibernate 5's javadoc mention a `QueryEngine` that should be used instead. We need to investigate how to properly update this code to use the new available APIs.

### Version

quarkus3 branch

### Expected behavior

Usage of native query apis should be migrated

### Actual behavior

Code is currently commented out

### How to Reproduce?

Use branch quarkus 3

**Comment by ahus1** [comment]
I analyzed the code with a focus on the JPA Map storage; there I added a native query for CockroachDB. 

The way I understand the current calls to `sessionFactory.getQueryPlanCache()` after some debugging: They trigger generating the query plan explicitly, while this would be done implicitly on the first use of the query. 

So it is safe to remove them, and should change any test results. I'd even argue to remove the lines completely to reduce the Hibernate API usage. 

This keeps me wondering why this was added in the first place. The two possibilities I see: 

* **Performance Optimization**: do this at startup, instead of at runtime later. Still I don't see the benefit here / how much of an improvement this would be.
* **Fail Fast**: Creating the query plan might throw an exception on initialization instead of only at query execution. This works for HQL, but not for native queries.

So my analysis: it is ok to remove the lines from `JpaMapUtil` and `JpaUtil` for now. We might later re-add them, or remove them.

Today, I'd be tending towards removing them if there is no really simple way to trigger the validation of the SQL.
